### PR TITLE
feat: answerapi-generated-answer

### PR DESCRIPTION
**Pull request by pathiery**
[SVCC-3790](https://coveord.atlassian.net/browse/SVCC-3790)

Create an atomic component that fronts the new answer api generated answer controller

[SVCC-3790]: https://coveord.atlassian.net/browse/SVCC-3790?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmgauthier** [comment]
Good luck with the CI.

**Review by dmgauthier — APPROVED** [review]

File: packages/headless/src/features/result-preview/result-preview-insight-analytics-actions.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const logDocumentQuickview = (result: Result): ClickAction => {
     analyticsPayloadBuilder: (state): InsightPanel.ItemAction => {
       const information = partialDocumentInformation(result, state);
       return {
-        action: 'preview',
+        action: 'preview' as unknown as InsightPanel.ItemAction['action'],
```
Comment (alexprudhomme) [inline packages/headless/src/features/result-preview/result-preview-insight-analytics-actions.ts:26]: Why is this needed ?

File: packages/atomic/src/components/knowledge/atomic-answerapi-generated-answer/atomic-answerapi-generated-answer.tsx [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    });
+    this.generatedAnswer = buildGeneratedAnswer(this.bindings.engine, {
+      initialState: {
+        isVisible: this.generatedAnswerCommon.data.isVisible,
+        responseFormat: {
+          answerStyle: this.answerStyle,
+          contentFormat: ['text/markdown', 'text/plain'],
+        },
+      },
+      answerConfigurationId: this.answerConfigurationId,
```
Comment (alexprudhomme) [inline packages/atomic/src/components/knowledge/atomic-answerapi-generated-answer/atomic-answerapi-generated-answer.tsx:153]: I find it somewhat redundant to duplicate the entire `atomic-generated-answer` component just to add this single line. What was the reasoning behind this ? Would it not have been simpler to add an option to the existing atomic-`atomic-generated-answer` component?

File: packages/headless/src/features/result-preview/result-preview-insight-analytics-actions.ts (hunk 1 above)
Comment (pathiery) [reply packages/headless/src/features/result-preview/result-preview-insight-analytics-actions.ts:26]: doesn't build without, lemme try it again

File: packages/atomic/src/components/knowledge/atomic-answerapi-generated-answer/atomic-answerapi-generated-answer.tsx (hunk 2 above)
Comment (alexprudhomme) [reply packages/atomic/src/components/knowledge/atomic-answerapi-generated-answer/atomic-answerapi-generated-answer.tsx:153]: After a dm discussion, my opinion still stands. 
We currently have a public headless controller diverging to 2 APIs via a prop. Why shouldn't we do the same thing on the atomic side and reuse the `AtomicGenratedAnswer` ?

Comment (louis-bompart) [reply packages/atomic/src/components/knowledge/atomic-answerapi-generated-answer/atomic-answerapi-generated-answer.tsx:153]: @tedre191, that ain't what we chatted about. :suspect:

Comment (fbeaudoincoveo) [reply packages/atomic/src/components/knowledge/atomic-answerapi-generated-answer/atomic-answerapi-generated-answer.tsx:153]: Agreed, if this is the only change then I don't see the need to duplicate the entire component.

We can instead add an optional prop on the `atomic-generated-answer` component and do:

```typescript
      ...(this.answerConfigurationId && {
        answerConfigurationId: this.answerConfigurationId,
      }),
```

here

**Review by alexprudhomme — APPROVED** [review]

**Review by dmgauthier — APPROVED** [review]

**Review by louis-bompart — APPROVED** [review]
oui

**Comment by louis-bompart** [comment]
Can be done in another J#/PR:
 - Add Storybook example/test

**Review by lbergeron — APPROVED** [review]
